### PR TITLE
Run3-alca191 Stop using cout in the codes of Calibration/HcalCalibAlgos

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Stop using cout in the codes of Calibration/HcalCalibAlgos

#### PR validation:

Use the runTheMatrix test workflows

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

File: Calibration/HcalCalibAlgos/plugins/Analyzer_minbias.cc [hunk 1]
```diff
@@ -390,7 +390,7 @@ namespace cms {
  
       if (!gtRecord.isValid()) {
 
-      //     LogDebug("L1GlobalTriggerRecordProducer")
+      //     edm::LogVerbatim("AnalyzerMB")
       //       << "\n\n Error: no L1GlobalTriggerReadoutRecord found with input tag "
       //       << m_l1GtReadoutRecord
       //       << "\n Returning empty L1GlobalTriggerRecord.\n\n";
```
Comment (tvami) [inline Calibration/HcalCalibAlgos/plugins/Analyzer_minbias.cc:396]: Please remove commented out codes, or explain why there are needed.

File: Calibration/HcalCalibAlgos/plugins/HOCalibAnalyzer.cc [hunk 2]
```diff
@@ -982,7 +982,7 @@ void HOCalibAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& i
   //swapped phi map for R0+/R0- (15/03/07)  
   for (int ij=0; ij<4; ij++) {
     for (int jk=0; jk<21; jk++) {
-      edm::LogInfo("HOCalib") <<"ieta "<<ij<<" "<<jk<<" "<<etamap[ij][jk];
+      edm::LogVerbatim("HOCalib") <<"ieta "<<ij<<" "<<jk<<" "<<etamap[ij][jk];
     }
   }
```
Comment (tvami) [inline Calibration/HcalCalibAlgos/plugins/HOCalibAnalyzer.cc:987]: Please remove commented out codes, or explain why there are needed.

File: Calibration/HcalCalibAlgos/plugins/HOCalibAnalyzer.cc [hunk 3]
```diff
@@ -1100,7 +1100,7 @@ void HOCalibAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& i
         for (int ij = 0; ij < 9; ij++) {
           hbhesig[ij] = (*hoC).hbhesig[ij];
         }
-      }  // edm::LogInfo("HOCalib")<<"hbhesig "<<ij<<" "<<hbhesig[ij];}}
+      }  // edm::LogVerbatim("HOCalib")<<"hbhesig "<<ij<<" "<<hbhesig[ij];}}
```
Comment (tvami) [inline Calibration/HcalCalibAlgos/plugins/HOCalibAnalyzer.cc:1103]: Please remove commented out codes, or explain why there are needed.

File: Calibration/HcalCalibAlgos/plugins/HOCalibAnalyzer.cc [hunk 4]
```diff
@@ -2176,7 +2177,7 @@ void HOCalibAnalyzer::endJob() {
                   if (step[k] > -10) {
                     gMinuit->mnpout(k, chnam, parv, err, xlo, xup, iuit);
                     gMinuit->mnerrs(k, plerr, mierr, eparab, gcc);
-                    //		    edm::LogInfo("HOCalib") <<"k "<< k<<" "<<chnam<<" "<<parv<<" "<<err<<" "<<xlo<<" "<<xup<<" "<<plerr<<" "<<mierr<<" "<<eparab;
+                    //		    edm::LogVerbatim("HOCalib") <<"k "<< k<<" "<<chnam<<" "<<parv<<" "<<err<<" "<<xlo<<" "<<xup<<" "<<plerr<<" "<<mierr<<" "<<eparab;
```
Comment (tvami) [inline Calibration/HcalCalibAlgos/plugins/HOCalibAnalyzer.cc:2180]: Please remove commented out codes, or explain why there are needed.

File: Calibration/HcalCalibAlgos/test/HcalIsoTrackStudy.cc [hunk 5]
```diff
@@ -83,6 +83,8 @@
 
 #include "SimDataFormats/GeneratorProducts/interface/GenEventInfoProduct.h"
 
+//#define EDM_ML_DEBUG
```
Comment (tvami) [inline Calibration/HcalCalibAlgos/test/HcalIsoTrackStudy.cc:86]: Please remove commented out codes, or explain why there are needed.

**Review by tvami — CHANGES_REQUESTED** [review]
Hi @bsunanda , please remove commented out codes, or explain why there are needed. Thanks!

File: Calibration/HcalCalibAlgos/test/HcalIsoTrackStudy.cc (hunk 5 above)
Comment (bsunanda) [reply Calibration/HcalCalibAlgos/test/HcalIsoTrackStudy.cc:86]: I shall keep this one. I need it to test sometimes with only one class

**Comment by perrotta** [comment]
please test

**Comment by tvami** [comment]
@perrotta is the bot stuck? It seems it couldn't perform tests in 13 hours... Maybe we should abort and reissue tests?

**Comment by tvami** [comment]
@cmsbuild , please abort

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by tvami** [comment]
+alca

File: Calibration/HcalCalibAlgos/plugins/AnalyzerMinbias.cc [hunk 6]
```diff
@@ -56,13 +59,15 @@
...
 // constructors and destructor
-class AnalyzerMinbias : public edm::EDAnalyzer {
+class AnalyzerMinbias : public edm::one::EDAnalyzer<edm::one::WatchRuns, edm::one::SharedResources> {
 public:
   explicit AnalyzerMinbias(const edm::ParameterSet&);
   ~AnalyzerMinbias() override;
 
   void analyze(const edm::Event&, const edm::EventSetup&) override;
   void beginJob() override;
+  void beginRun(edm::Run const&, edm::EventSetup const&) override {}
```
[the remark this answers is not in the document]
Comment (bsunanda) [reply Calibration/HcalCalibAlgos/plugins/AnalyzerMinbias.cc:69]: Keeping a placeholder for future extension

File: Calibration/HcalCalibAlgos/plugins/HOCalibAnalyzer.cc [hunk 7]
```diff
@@ -203,14 +200,16 @@ void set_sigma(double& x, bool mdigi) {
...
-class HOCalibAnalyzer : public edm::EDAnalyzer {
+class HOCalibAnalyzer : public edm::one::EDAnalyzer<edm::one::WatchRuns, edm::one::SharedResources> {
 public:
   explicit HOCalibAnalyzer(const edm::ParameterSet&);
   ~HOCalibAnalyzer() override;
 
 private:
   void beginJob() override;
   void analyze(const edm::Event&, const edm::EventSetup&) override;
+  void beginRun(edm::Run const&, edm::EventSetup const&) override {}
```
[the remark this answers is not in the document]
Comment (bsunanda) [reply Calibration/HcalCalibAlgos/plugins/HOCalibAnalyzer.cc:211]: The same story - there may be run dependent action

File: Calibration/HcalCalibAlgos/plugins/AnalyzerMinbias.cc (hunk 6 above)
[the remark this answers is not in the document]
Comment (perrotta) [reply Calibration/HcalCalibAlgos/plugins/AnalyzerMinbias.cc:69]: I don't think we should foresee placeholders with empty methods for every possible future use case.
If sometime in the future some run-dependent action will have to be implemented in the code, we will have to modify it anyhow: therefore, there is no advantage in having them here defined as such.

[the remark this answers is not in the document]
Comment (bsunanda) [reply Calibration/HcalCalibAlgos/plugins/AnalyzerMinbias.cc:69]: OK I shall take care of these

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@tvami @perrotta Could you approve and integrate this PR now?

**Comment by tvami** [comment]
+alca

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by qliphy** [comment]
+1

**Pull request merged**